lex/lexer.go

```diff
@@ -2623,7 +2554,7 @@ func LexExpression(l *Lexer) StateFn {
 		}
 	}
 	// ensure we don't get into a recursive death spiral here?
-	if len(l.stack) < 1000 {
+	if len(l.stack) < l.depthLimit || l.depthLimit <= -1 {
```

Comment:
personally i think < 0 is more idiomatic of this sort of thing, and that theoretically allowing setting it to 0 might be useful in some cases, like for testing. not a big deal though